src/ring_theory/hahn_series.lean

```diff
@@ -719,4 +733,250 @@ dif_neg hx
 
 end valuation
 
+section
+variables (Γ) (R) [linear_order Γ] [add_comm_monoid R]
+
+/-- An infinite family of Hahn series which has a formal coefficient-wise sum.
+  The requirements for this are that the union of the supports of the series is well-founded,
+  and that only finitely many series are nonzero at any given coefficient. -/
+structure summable_family (α : Type*) :=
+(to_fun : α → hahn_series Γ R)
+(is_wf_Union_support' : set.is_wf (⋃ (a : α), (to_fun a).support))
+(co_support : Γ → finset α)
+(mem_co_support' : ∀ (a : α) (g : Γ), a ∈ co_support g ↔ (to_fun a).coeff g ≠ 0)
+
+end
+
+namespace summable_family
+section add_comm_monoid
+
+variables [linear_order Γ] [add_comm_monoid R] {α : Type*}
+
+instance : has_coe_to_fun (summable_family Γ R α) :=
+⟨λ _, (α → hahn_series Γ R), to_fun⟩
+
+lemma is_wf_Union_support (s : summable_family Γ R α) : set.is_wf (⋃ (a : α), (s a).support) :=
+s.is_wf_Union_support'
+
+@[simp]
+lemma mem_co_support {s : summable_family Γ R α} {a : α} {g : Γ} :
+  a ∈ s.co_support g ↔ (s a).coeff g ≠ 0 := mem_co_support' _ _ _
+
+@[ext]
+lemma ext {s t : summable_family Γ R α} (h : ∀ (a : α), s a = t a) : s = t :=
```

Comment:
Can you also prove the following?
```lean
lemma coe_injective : function.injective (coe_fn : summable_family Γ R α → α → hahn_series Γ R)
```
It should be straightforward to prove `ext` from `coe_injective` or vice-versa.

Reply:
It doesn't let me do that, it complains that `coe_fn` has type `Π (x : summable_family Γ R α), has_coe_to_fun.F x`

Reply:
Perhaps the following then?
```lean
lemma coe_injective : @function.injective (summable_family Γ R α) (α → hahn_series Γ R) coe_fn
```
This problem came up in `finsupp` too.